WalletWasabi/Services/IndexDownloader.cs

```diff
@@ -158,33 +160,28 @@ public void Synchronize(TimeSpan requestInterval)
 
 								continue;
 							}
-							else if (response.StatusCode == HttpStatusCode.NotFound)
+						}
+						catch (HttpRequestException ex) when (bestKnownFilter != null && ex.Message.StartsWith("Not Found\nProvided bestKnownBlockHash is not found:"))
```

Comment:
Don't take flow decision based on strings. I don't know if there is a better way here but at leat you could do:
```c# 
ex.Message.StartsWith( HttpStatusCode.NotFound.ToReasonString() )
```

Q: Why is this comparison `bestKnownFilter != null` for? Is it possible to be null? In case it can be null, what doe it means?

Reply:
Done.

I added the null check because I was moved the declaration outside the try/catch and was worried that the filter could have been null, but as its value is assigned before `GetFiltersAsync` is called, it will have a value. So removed as it's redundant.